updater/src/Updater/Generate/Template.purs

```diff
@@ -154,75 +146,90 @@ runTemplate opts (Template { from, to }) = do
 
   FS.Extra.writeTextFile to (replaceVariables opts.variables templateContents)
 
+-- | Keep JS only templates when using JS.
+filterByType :: Boolean -> Array TemplateSource -> Array TemplateSource
+filterByType true templates = templates
+filterByType false templates = filter (not <<< (JS == _) <<< sourceType) templates
+
+-- | Define the source and destination of a given template by its type and
+-- | source path. Common templates default to _base_ unless using JS.
+templateFromSource :: Boolean -> TemplateSource -> Template
+templateFromSource true (TemplateSource Common path) =
+  Template { from: "js/" <> path, to: path }
+templateFromSource false (TemplateSource Common path) =
+  Template { from: "base/" <> path, to: path }
+templateFromSource _ (TemplateSource JS path) =
+  Template { from: "js/" <> path, to: path }
+templateFromSource _ (TemplateSource Base path) =
+  Template { from: "base/" <> path, to: path }
+
+-- | Validate that the given paths of templates to run exist and for JS
+-- | templates that only are provided together with '--uses-js'
+validateFiles :: Boolean -> Array TemplateSource -> NonEmptyList FilePath -> Either String (Array TemplateSource)
+validateFiles usesJS templates files = fromFoldable <$> traverse validateFile files
+  where
+  validateFile :: FilePath -> Either String TemplateSource
+  validateFile path =
+    case find ((path == _) <<< sourcePath) templates of
+      Nothing -> Left $ "Path '" <> path <> "' is not a valid template"
+      Just (TemplateSource JS _) | not usesJS ->
+        Left $ "Path '" <> path <> "' is a JS only template. Did you forget '--uses-js'?"
+      Just template -> Right template
+
+data TemplateSourceType = Base | JS | Common
+
+derive instance eqTemplateSourceType :: Eq TemplateSourceType
+
 -- | The source and destination for a given template. Templates will be copied
 -- | from their source to destination, with text replacement applied along the way.
 newtype Template = Template { from :: FilePath, to :: FilePath }
```

Comment:
We can remove this type now that we're using the `TemplateSourceType`. Instead of having a dedicated type like this we can just have a `formatPath` or something that goes from `TemplateSource -> FilePath` and constructs the file path you need.